pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -61,6 +70,103 @@ func (r *ReconcileStorageCluster) ensureStorageClasses(instance *ocsv1.StorageCl
 	return nil
 }
 
+// ensureExternalClusterConfig ensures that Storagclass for the external cluster exist
+func (r *ReconcileStorageCluster) ensureExternalClusterConfig(instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	// check for the status boolean value accepted or not
+	if instance.Status.ExternalSecretFound {
+		return nil
+	}
+	found := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rook-ceph-external-cluster-details",
+			Namespace: "openshift-storage",
+		},
+
+		Data: map[string][]byte{
+			"external_cluster_details": []byte(""),
+		},
+	}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: found.Name, Namespace: found.Namespace}, found)
+	if err != nil { // external cluster already found
+		return err
+	}
+	ownerRef := metav1.OwnerReference{
+		UID:        instance.UID,
+		APIVersion: instance.APIVersion,
+		Kind:       instance.Kind,
+		Name:       instance.Name,
+	}
+	var data ExternalResources
+	err = json.Unmarshal([]byte(found.Data["external_cluster_details"]), &data)
+	if err != nil {
+		return fmt.Errorf("could not parse json blob, got %+v", err)
+	}
+	for _, d := range data {
+		switch d.Kind {
+		case "ConfigMap":
+			cm := &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            d.Name,
+					Namespace:       d.Name,
+					OwnerReferences: []metav1.OwnerReference{ownerRef},
+				},
+			}
+			err = controllerutil.SetControllerReference(instance, cm, r.scheme)
+			if err != nil {
+				return err
+			}
+			_, err = controllerutil.CreateOrUpdate(context.TODO(), r.client, cm, func() error {
+				cm.Data = d.Data
+				return nil
+			})
+			if err != nil && !errors.IsAlreadyExists(err) {
+				return err
+			}
+		case "StorageClass":
```

Comment:
The StorageClass case needs to be adapted to reflect with the existing logic to create SC. The data has no meaning here. The only things that matter here are:

* for cephfs: `d.FilesystemName`
* for rbd: `d.PoolName`
* for obc: `d.Data['endpoint']`

Reply:
I believe that this is where the "names" will come for the corresponding Storageclass?.
Can you please give more details that if we don't have to use Switch case for storage class creation then what?
I couldn't come up with another logic. Thanks

Reply:
ocs-op already has a logic to create the rbd and cephfs pool, just re-use it and add a new one for the OBC.

Reply:
I can't see what the code was, but in general I thought the idea was to just blindly create whatever data we're given. In which case there should no need to integrate any of this information in other parts of the code.

Reply:
@leseb Wait, so the `d.Data` field contains nothing of value for the StorageClass?? Why aren't the parameters there? They're just a `map[string]string` like `d.Data` is. Are we expected to distinguish what kind of parameters to set based on whether certain fields are non-null?

Reply:
@leseb Wait, so the `d.Data` field doesn't contain the `spec.Parameters` value for the StorageClass? Why not? They're just a `map[string]string` like `d.Data` is. Are we expected to distinguish what kind of parameters to set based on whether certain fields are non-null?

Reply:
Well since you do have the parameters already I don't see why they should be coming from the JSON blob. The blob reflects what the external env is, there is no concept of SC there.

Because the SC creation logic is already present we just need to slightly re-adapt it whether this is an external or converged cluster.

Reply:
...okay, but why are some of the parameters explicit fields and only one of them is a `map[string]string`?

Reply:
So that we don't alter the map. We thought it'd make a clearer difference for Secret and ConfigMap `Data` because the JSON actually builds the `cm.Data`. So you can blindly copy it.

We only make a difference for the SC, which still has a Data field, but I see your point, for an SC we can rename it to `Parameters` if you prefer but still you would have to complete the rest of it. Let us know soon.